model/types.go

```diff
@@ -76,63 +80,37 @@ func (h Header) TableName() string {
 
 // AddReward adds reward to header.
 // Verify that there are at most 2 uncles
-func (h Header) AddReward(txsFee, minerBaseReward, uncleInclusionReward *big.Int, unclesReward []*big.Int, unclesHash []common.Hash) (*Header, error) {
-	if len(unclesReward) != len(unclesHash) {
+func (h Header) AddReward(txsFee, minerBaseReward, uncleInclusionReward *big.Int, unclesReward []*big.Int, uncleCBs []common.Address, unclesHash []common.Hash) (*Header, error) {
+	if len(unclesReward) != len(unclesHash) || len(unclesReward) != len(uncleCBs) {
 		return nil, ErrConfusedUncles
 	}
 	if len(unclesReward) > maxUncles {
 		return nil, ErrTooManyUncles
 	}
-	totalUnclesReward := new(big.Int)
+	// We assume that the length of  coinbases, rewards and hashes are the same.
+	urd := make([]*big.Int, maxUncles)
 	ush := make([][]byte, maxUncles)
+	ucb := make([][]byte, maxUncles)
 	for i, u := range unclesHash {
+		urd[i] = unclesReward[i]
 		ush[i] = u.Bytes()
-		totalUnclesReward.Add(totalUnclesReward, unclesReward[i])
+		ucb[i] = uncleCBs[i].Bytes()
 	}
 	minerReward := new(big.Int).Add(txsFee, minerBaseReward)
 	minerReward.Add(minerReward, uncleInclusionReward)
 
 	h.MinerReward = minerReward.String()
 	h.UnclesInclusionReward = uncleInclusionReward.String()
 	h.TxsFee = txsFee.String()
-	h.UnclesReward = totalUnclesReward.String()
+	h.Uncle1Reward = urd[0].String()
```

Comment:
GO question. if there is no uncle. will we have a exception for null memory access?

Reply:
No, because I made a list of length 2 https://github.com/getamis/eth-indexer/pull/55/files/13f3b84999b517cf2798cfb8dd7c07f5ec28c295#diff-7eff12030952ecb8157e47a32a5d8664R91

Reply:
and each element have default value(empty), so the null exception doesn't happen